deck/card.py

```diff
@@ -1,43 +1,80 @@
-# Definição da classe
-class Card:
-
-    # Membros estáticos (coletivos) da classe
-    # Todos os objetos podem acessá-los
-    __names = {1 : "ace",
-               2 : "two",
-               3 : "three",
-               4 : "four",
-               5 : "five",
-               6 : "six",
-               7 : "seven",
-               8 : "eight",
-               9 : "nine",
-               10 : "ten",
-               11 : "queen",
-               12 : "jack",
-               13 : "king"}
-
-    # Construtor
-    def __init__(self, value, suit):
-        self.__value = value
-        self.__suit = suit
-
-        # Verificando o argumento 'value'
-        if type(self.__value) == int:
-            if self.__value < 1 or self.__value > 13: 
+from abstract_card import AbstractCard
+
+# Class to represent a card from a common deck
+class Card(AbstractCard):
+
+    # Defining names of the values of the cards
+    __NAMES = {1 : "Ace",
+               2 : "Two",
+               3 : "Three",
+               4 : "Four",
+               5 : "Five",
+               6 : "Six",
+               7 : "Seven",
+               8 : "Eight",
+               9 : "Nine",
+               10 : "Ten",
+               11 : "Queen",
+               12 : "Jack",
+               13 : "King"}
+    
+    # Defining suits
+    __SUITS = ("Hearts", "Diamonds", "Clubs", "Spades")
+    __COLORS = ("Blue", "Red")
```

Comment:
Corres não podem ser inferidos pelo naipe? Copa e ouros vermelho, e bastos e espada preto(azul)

Reply:
The variable "__COLORS" is referent the color of the back side of the card. Not the color of the suits.

Reply:
uhmm, but this can be a property from the deck, since in a full deck you cannot have different back colors. It´s more a characteristic from deck, even if in real world this color is from a card. With this, you avoid duplicity of this information.